core/src/main/scala/kafka/server/KafkaRaftServer.scala

```diff
@@ -176,4 +180,8 @@ object KafkaRaftServer {
     (metaProperties, offlineDirs.toSeq)
   }
 
+  val configSchema = new KafkaConfigSchema(Map(
+    ConfigResource.Type.BROKER -> KafkaConfig.configDef,
+    ConfigResource.Type.TOPIC -> LogConfig.configDefCopy,
```

Comment:
I realize this is just copied over from ControllerServer, but why do we make a copy of the LogConfig but not KafkaConfig here?

Reply:
In practice nobody mutates either of these. But you're right, it is inconsistent. I will do a defensive copy on the KafkaConfig one.

In the future we should have a freeze method or something on the ConfigDef to make these defensive copies unnecessary. But that's out of scope here...